modules/ROOT/pages/cors-policy.adoc

```diff
@@ -14,20 +14,125 @@ endif::[]
+
+SOP (same-origin policy) deters malicious attackers from exploiting cookies when one web page invokes another. For example, you might log in to your bank's web page using a cookie that an attacker might be able to obtain and exploit to query the bank’s API on your behalf. 
+
+[[origin]]
+With the SOP, scripts can access data from a target web page only if it has the same origin as the caller web page. For more information about origins, see https://en.wikipedia.org/wiki/Same-origin_policy[Wikipedia].
+
+The following example illustrates the origin of the web page, http://www.example.com:8080:
+
+image:cors-policy-sop-example.png][SOP Example,95%,85%]
+
+Because the SOP is highly restrictive, access from one suborigin to another or to external hyperlinks do not work on web pages. For example, if an origin `www.testapply.com` has two suborigins, `www.eng.testapply.com` and `www.docs.testapply.com`, communication between the two suborigins is denied. Additionally, any hyperlinks to external websites from any of the suborigins is also denied.
+
+To circumvent this problem, web browsers implemented the CORS standard, which validates the web server and accepts the request if the validation is successful.
+
+For example, if your bank implements the CORS server-side protocol on its login server, you can query data only directly from the bank's web page. Any attempt to query the login API from an external (non-bank) domain is denied.
+
+== How CORS Works
+
+When your web pages request data, the browser detects whether the request is from within same origin and determines whether to apply the CORS algorithm. If you query data from a web page that is not in your origin, then the CORS policy is applied. 
+
+The CORS algorithm works on the web server and on the client-side for the web page that requested the information. The client-side algorithm in the CORS policy is implemented by:
+
+* Determining whether the request is complex (and potentially dangerous) and sending a preliminary _preflight_ request to verify whether the server accepts the origin.
+* Executing the actual request and validating that the server responds correctly and accepts the origin.
+
+A _preflight_ is a preliminary request (using OPTIONS as the HTTP method) from the web browser to the backend server to test the identity (origin and a few other headers) of the web page that is trying to perform the request. 
+
+If the backend does not accept the origin, the backend server responds to the request without a specific header (Access-Control-Allow-Origin). The client then understands that the page’s origin is not allowed in that server and does not execute the actual request.
+
+The following diagram shows the XMLHttpRequest (XHR) in a JavaScript flow for determining whether to execute the actual request:
+
+image::cors-policy-xhr-diagram.png[]
+
+When the request is executed and it reaches the backend, the server side of the algorithm is initiated in two steps:
```

Comment:
This is actually a summary of the interaction between backend and browser.
Change this for smth like "The detail interaction between the browser and the server goes as follows:"

Reply:
done! Thanks!